Content.Server/Radio/EntitySystems/MessagesServerSystem.cs

```diff
@@ -0,0 +1,129 @@
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Content.Server.CartridgeLoader.Cartridges;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Robust.Shared.Timing;
+using Content.Shared.CartridgeLoader;
+using Content.Server.CartridgeLoader;
+using Content.Server.Radio.Components;
+using Robust.Shared.Containers;
+using Content.Server.Power.Components;
+using Content.Server.Power.EntitySystems;
+using System.Linq;
+using Content.Server.DeviceNetwork;
+using Content.Server.DeviceNetwork.Components;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Shared.DeviceNetwork;
+
+
+namespace Content.Server.Radio.EntitySystems;
+
+public sealed class MessagesServerSystem : EntitySystem
+{
+
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly MessagesCartridgeSystem _messagesCartridgeSystem = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+    [Dependency] private readonly SingletonDeviceNetServerSystem _singletonServerSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<MessagesServerComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+    }
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        var serverQuery = EntityQueryEnumerator<MessagesServerComponent>();
+        while (serverQuery.MoveNext(out var uid, out var server))
+        {
+            if (!_singletonServerSystem.IsActiveServer(uid))
+                continue;
+            if (server.NextUpdate <= _gameTiming.CurTime)
+            {
+                server.NextUpdate += server.UpdateDelay;
+
+                Update(uid, server);
+            }
+        }
+    }
+
+    public void Update(EntityUid uid, MessagesServerComponent component)
+    {
+        if (!this.IsPowered(uid, EntityManager))
+            return;
+
+        var serverQuery = EntityManager.AllEntityQueryEnumerator<MessagesServerComponent>();
+
+        while (serverQuery.MoveNext(out var serverUid, out var serverComponent))
+        {
+            component.Messages = new List<MessagesMessageData>(component.Messages.Union(serverComponent.Messages));
+            serverComponent.Messages = new List<MessagesMessageData>(component.Messages);
+        }
+
```

Comment:
Instead of just doing copies to every single server can't you just store the messages on some abstract entity instead? It will be hundreds of times faster than what this code is doing.